theme/static/js/profile.js

```diff
@@ -245,6 +245,56 @@ function getUrlVars()
     return vars;
 }
 
+function isPhoneValidInCountry(phone, country){
+    if (!BFHPhoneFormatList.hasOwnProperty(country)) return false;
+    let BFHString = BFHPhoneFormatList[country];
+
+    // Turn the string into a regex
+    stringFormat = BFHString.replace(/\s/g, String.fromCharCode(92)+"s");
+    stringFormat = stringFormat.replace("+", String.fromCharCode(92)+"+");
+    stringFormat = stringFormat.replace("(", String.fromCharCode(92)+"(");
+    stringFormat = stringFormat.replace(")", String.fromCharCode(92)+")");
+    stringFormat = stringFormat.replace(/d/g, String.fromCharCode(92)+"d");
+    stringFormat = new RegExp(stringFormat);
+    return stringFormat.test(phone) || BFHString;
+}
+
+function isPhoneCountryCodeOnly(phone, country){
+    if (!BFHPhoneFormatList.hasOwnProperty(country)) return false;
+    let BFHString = BFHPhoneFormatList[country];
+    let stringFormat = BFHString.replace(/d/g, '');
+    stringFormat = stringFormat.replace(/[\(\)-]/g, '');
+    return phone.trim().length === stringFormat.trim().length
+}
+
+function resetPhoneValues(){
+    const oldPhones = PHONES || null;
+    if (!oldPhones || oldPhones === 'None') return;
+    $('input[type="tel"]').each(function(){
+        let phoneField = $(this);
+        if (oldPhones[phoneField[0].name] === 'None') return;
+        phoneField.val(oldPhones[phoneField[0].name]);
+    });
+    checkForInvalidPhones();
+}
+
+function checkForInvalidPhones(country){
+    country = country || $("#country").val()
```

Comment:
I think the code below should only execute if `country` has a value. Otherwise we get this uncaught error:
![image](https://user-images.githubusercontent.com/2448568/211387661-8d95cb41-b67c-4ebc-b302-99aa8febe706.png)

Reply:
thanks I hadn't noticed this.
I wonder if this is a bug with bootstrap-formhelpers, because it seems to occur any time there is a `bfh-phone` field in the form that has `data-country` attribute targeting a `bfh-countries` select element that doesn't have a default value.

I tried checking for country=null as you suggested, but this TypeError is thrown on page load, before our [document.ready() callback](https://github.com/hydroshare/hydroshare/blob/4235-country-codes_s/theme/static/js/profile.js#L298)

I'll have to think more about a workaround, or maybe scrap bfh...
Or, since "Country" is a required field on signup and profile, shouldn't it always have a value? (and we wouldn't have to worry about this bfh error)?
I only get an empty Country if I delete it direct in my database, or if I use one of the users in our dev db that doesn't have a country set...

Reply:
ah NVM, I see now that we have a lot of users in our db that don't have "country" in their profile (I guess from before we started requiring it).

```
>>> from theme.models import UserProfile
>>> UserProfile.objects.filter(country=None).all()
<QuerySet [<UserProfile: UserProfile object (183)>, <UserProfile: UserProfile object (14)>, <UserProfile: UserProfile object (16)>, <UserProfile: UserProfile object (19)>, <UserProfile: UserProfile object (22)>, <UserProfile: UserProfile object (27)>, <UserProfile: UserProfile object (29)>, <UserProfile: UserProfile object (31)>, <UserProfile: UserProfile object (32)>, <UserProfile: UserProfile object (33)>, <UserProfile: UserProfile object (34)>, <UserProfile: UserProfile object (36)>, <UserProfile: UserProfile object (38)>, <UserProfile: UserProfile object (39)>, <UserProfile: UserProfile object (40)>, <UserProfile: UserProfile object (42)>, <UserProfile: UserProfile object (43)>, <UserProfile: UserProfile object (44)>, <UserProfile: UserProfile object (49)>, <UserProfile: UserProfile object (51)>, '...(remaining elements truncated)...']>

Okay, I will revamp this PR, thanks @Maurier!!!

Reply:
@Maurier I decided to keep bootstrap-form-helper and implement a fix directly within our copy of it.
The repo looks like it hasn't been updated since 2018, so I'm not to keen on submitting a PR to https://github.com/winmarkltd/BootstrapFormHelpers